numpy/ma/tests/test_core.py

```diff
@@ -1245,6 +1245,49 @@ def test_minmax_methods(self):
         assert_(x.max() is masked)
         assert_(x.ptp() is masked)
 
+    def test_minmax_dtypes(self):
+        # Additional tests on max/min for non-standard float and complex dtypes
+        x = np.array([1., 1., 1., -2., pi/2.0, 4., 5., -10., 10., 1., 2., 3.])
+        a10 = 10.
+        an10 = -10.0
+        m1 = [1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]
+        xm = masked_array(x, mask=m1)
+        xm.set_fill_value(1e+20)
+        float_dtypes = [np.half, np.single, np.double,
+                        np.longdouble, np.cfloat, np.cdouble, np.clongdouble]
+
+        def min_max_dtypes(float_dtype):
+            assert_equal(masked_array(x, mask=m1, dtype=float_dtype).max(),
+                         float_dtype(a10))
+            assert_equal(masked_array(x, mask=m1, dtype=float_dtype).min(),
+                         float_dtype(an10))
+
+        for float_dtype in float_dtypes:
+            min_max_dtypes(float_dtype)
+
+        assert_equal(xm.min(), an10)
+        assert_equal(xm.max(), a10)
+
+        # Non-complexs type only test
+
+        def min_max_nocomplex(float_dtype):
+            ym = masked_array([15e19, 2e20, 16e19], mask=[1, 0, 0],
+                              dtype=float_dtype)
+            assert_equal(ym.min(), float_dtype(16e19))
+            assert_equal(ym.max(), float_dtype(2e20))
+        for float_dtype in float_dtypes[:2]:
+            min_max_nocomplex(float_dtype)
+
+        # Complex types only test
+        ym = masked_array([1e20+1j, 1e20-2j, 1e20-1j], mask=[0, 1, 0],
+                          dtype=np.cdouble)
+        assert_equal(ym.min(), np.cdouble(1e20-1j))
+        assert_equal(ym.max(), np.cdouble(1e20+1j))
+        ym = masked_array([1e20+1j, 1e20-2j, 1e20-1j], mask=[0, 1, 0],
+                          dtype=np.clongdouble)
+        assert_equal(ym.min(), np.clongdouble(1e20-1j))
+        assert_equal(ym.max(), np.clongdouble(1e20+1j))
```

Comment:
shouldnt the max be (1e+20+1j) and min be (1e+20-2j)

Reply:
We mask the middle value, hence it is out of min max equation

Reply:
aah missed that ! thanks!